web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -268,18 +270,21 @@ function getTrustedSecurityGroups(
   securityGroups: SecurityGroup[],
   db: AwsRdsDatabase
 ): Set<string> {
-  const securityGroupsById = new Map(
-    securityGroups.map(group => [group.id, group])
-  );
+  const trustedGroups = new Set<string>();
+  if (!db) {
+    return trustedGroups;
+  }
 
   const dbPort = getPort(db);
-  const trustedGroups = new Set<string>();
-  db.securityGroups.forEach(groupId => {
+  const securityGroupsById = new Map(
+    securityGroups?.map(group => [group.id, group])
+  );
+  db?.securityGroups?.forEach(groupId => {
     const group = securityGroupsById.get(groupId);
     if (!group) {
       return;
     }
-    group.inboundRules.forEach(rule => {
+    group.inboundRules?.forEach(rule => {
```

Comment:
do we expect `inboundRules` to be undefined? i didn't see this being an optional field in the main data type `SecurityGroup`

Reply:
yeah good catch, the constructor will make sure it's not undefined. removed the `?`